### PR TITLE
feat: `experimental.viteMode`

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
* **#2849** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2849?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2849?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: crates/rolldown/tests/rolldown/topics/new_url/opt-out-vite-mode/_config.json
Comment (hyf0) [inline crates/rolldown/tests/rolldown/topics/new_url/opt-out-vite-mode/_config.json:1]: snake case.

**Review by hyf0 — APPROVED** [review]

File: crates/rolldown_common/src/inner_bundler_options/types/experimental_options.rs [hunk 1]
```diff
@@ -22,4 +26,13 @@ impl ExperimentalOptions {
   pub fn is_disable_live_bindings_enabled(&self) -> bool {
     self.disable_live_bindings.unwrap_or(false)
   }
+
+  #[inline]
+  pub fn get_ignore_comment(&self) -> &'static str {
+    if self.vite_mode.unwrap_or_default() {
+      "/* @vite-ignore */"
```
Comment (hyf0) [inline crates/rolldown_common/src/inner_bundler_options/types/experimental_options.rs]: NIT: I wonder should we be strict about spaces/patterns? Seems that the comment must be exact `/* @vite-ignore */`.

Comment (IWANABETHATGUY) [reply crates/rolldown_common/src/inner_bundler_options/types/experimental_options.rs]: I fixed the space issue, now including `@vite-ignore` should be enough.

**Pull request merged**